hydrogym/core.py

```diff
@@ -339,12 +387,31 @@ def __init__(self, env_config: dict):
         dtype=float,
     )
     self.action_space = gym.spaces.Box(
-        low=-self.flow.MAX_CONTROL,
-        high=self.flow.MAX_CONTROL,
+        low=self.flow.MAX_CONTROL_LOW,
+        high=self.flow.MAX_CONTROL_UP,
         shape=(self.flow.num_inputs,),
         dtype=float,
     )
 
+    self.t = 0.
+    self.dt = env_config.get("solver_config", {}).get("dt", None)
+    assert self.dt is not None, f"Error: Solver timestep dt ({self.dt}) must not be None"
+    self.num_sim_substeps_per_actuation = env_config.get(
+        "actuation_config", {}).get("num_sim_substeps_per_actuation", None)
+
+    if self.num_sim_substeps_per_actuation is not None and self.num_sim_substeps_per_actuation > 1:
+      assert self.rewardLogCallback is not None,\
+        f"Error: If num_sim_substeps_per_actuation ({self.num_sim_substeps_per_actuation}) " \
+        "is set a reward callback function must be given, {self.rewardLogCallback}"
+      self.reward_aggreation_rule = env_config.get("actuation_config", {}).get(
```

Comment:
what's the purpose of the `reward_aggregation_rule` in a `gym` interface where only instantaneous rewards/costs are considered? Is it the aggregations of the values in between decision times (times where the user hits `step` ).

Reply:
(there is a typo in the variable name `reward_aggreation_rule`)

Reply:
Could this all happen in some kind of [Wrapper](https://gymnasium.farama.org/api/wrappers/)?  All of this seems to add a lot of complexity to an otherwise straightforward class.

Reply:
Wrappers can deal with those aggregation rules and also action scaling.